pit30m/data/log_reader.py

```diff
@@ -318,24 +364,23 @@ def map_relative_poses_dense(self) -> np.ndarray:
     def utm_poses_dense(self) -> Tuple[np.ndarray, np.ndarray]:
         """UTM poses for the log, ordered by time.
 
-        TODO(andrei): Update to provide altitude.
-
         Returns:
             A tuple with two elements:
-              - An n-long boolean array indicating whether the poses are valid
-              - An n-by-2 array with the UTM xy coordinates of the poses
+                - An n-long boolean array indicating whether the poses are valid
+                - An n-by-3 array with the UTM xy coordinates and altitudes of the poses
         """
         mrp = self.map_relative_poses_dense
         xyzs = rfn.structured_to_unstructured(mrp[["x", "y", "z"]])
+
         # Handle submap IDs which were truncated upon encoded due to ending with a zero.
-        submaps = [UUID(bytes=submap_uuid_bytes.ljust(16, b"\x00")) for submap_uuid_bytes in mrp["submap_id"]]
+        submaps = [UUID(bytes=submap_uuid_bytes).ljust(16, b"\x00") for submap_uuid_bytes in mrp["submap_id"]]
```

Comment:
This looks like a typo. I think the previous code was correct.
```suggestion
        submaps = [UUID(bytes=submap_uuid_bytes.ljust(16, b"\x00")) for submap_uuid_bytes in mrp["submap_id"]]
```

I think adding a test (maybe in the integration tests of log reader) that calls `utm_poses_dense` from a log with nuked poses would flag this -- I already wrote this for another PR, happy to add it separately myself.

Reply:
Reverted & test added! Nice catch!